src/FsAutoComplete/LspServers/AdaptiveServerState.fs

```diff
@@ -156,6 +156,29 @@ type FindFirstProject() =
 
         $"Couldn't find a corresponding project for {sourceFile}. \n Projects include {allProjects}. \nHave the projects loaded yet or have you tried restoring your project/solution?")
 
+module TestProjectHelpers =
+  let tryGetWorkspaceProjects (workspaceLoader: IWorkspaceLoader) (workspace: WorkspaceChosen) =
+    match workspace with
+    | WorkspaceChosen.NotChosen -> Error "No workspace loaded. Can't discover tests"
+    | WorkspaceChosen.Projs projectPaths ->
+      projectPaths
+      |> List.ofSeq
+      |> List.map string
+      |> workspaceLoader.LoadProjects
+      |> Ok
```

Comment:
```suggestion

```

Instead of loading projects again, you'll want to reuse the currently loaded projects as this will help with any project related changes like adding files, adding packages etc.

https://github.com/ionide/FsAutoComplete/blob/20167d6d65aedc7f641cb27dd571c2d2c0830f50/src/FsAutoComplete/LspServers/AdaptiveServerState.fs#L944

You can do something like:

```fsharp
let! projects = projectOptions |> AsyncAval.forceAsync
let! testProjects = projectOptions.ToValueSeq() |> TestProjectHelpers.tryGetTestProjects
```

In DiscoverTests/RunTests